crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -931,11 +980,10 @@ impl<'db> KnownClass {
             | Self::MethodWrapperType
             | Self::WrapperDescriptorType => KnownModule::Types,
             Self::NoneType => KnownModule::Typeshed,
-            Self::SpecialForm
-            | Self::TypeVar
-            | Self::TypeAliasType
-            | Self::StdlibAlias
-            | Self::SupportsIndex => KnownModule::Typing,
+            Self::SpecialForm | Self::TypeVar | Self::StdlibAlias | Self::SupportsIndex => {
+                KnownModule::Typing
+            }
+            Self::TypeAliasType => KnownModule::TypingExtensions,
```

Comment:
The change here is that the canonical module for `KnownClass::TypeAliasType` is now `KnownModule::TypingExtensions` rather than `KnownModule::Typing`. Running the property tests on this branch instantly crashed without this change, as it was revealed that until now `KnownClass::TypeAliasType.to_instance(db)` has been silently resolving to `Type::Unknown` all this time, due to the fact that it was only added to the typing module in Python 3.12 but our default Python version is 3.9.